extensions/wikia/DataWarehouse/DataWarehouseEventProducer.class.php

```diff
@@ -375,10 +378,21 @@ public function setCategory() {
 		$this->mParams['categoryId'] = WikiFactory::getCategory( $this->app->wg->CityId );
 	}
 
+	public function setBeacon( $beacon ) {
+		$this->mParams['beacon'] = $beacon;
+	}
+
 	public function sendLog() {
 		wfProfileIn( __METHOD__ );
 		$data = json_encode($this->mParams);
 		$this->getRabbit()->publish( $this->mKey, $data );
+		if ( ( ! Wikia::isDevEnv()) && ( $this->mParams['cityId'] % 100 == 28 ) ) { // enabled on 1% of wikis
+			$this->mParams['action'] = $this->mKey;
+			$kinesis = new AsyncKinesisProducerTask();
+			( new AsyncTaskList() )
```

Comment:
use `newLocalTask` static constructor instead. AFAIR you need to specify wiki ID when queuing a task. This constructor will use the ID of the current wiki.

Reply:
OK, thanks!